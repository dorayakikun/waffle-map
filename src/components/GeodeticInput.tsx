import * as React from "react";
import { Dropdown, DropdownItemProps, DropdownProps } from "semantic-ui-react";

export type Props = {
  unit: string;
  datum: string;
  changeUnit: (
    event: React.SyntheticEvent<HTMLElement>,
    data: DropdownProps
  ) => void;
  onDatumChanged: (
    event: React.SyntheticEvent<HTMLElement>,
    data: DropdownProps
  ) => void;
};
const units: DropdownItemProps[] = [
  {
    text: "millisec",
    value: "millisec"
  },
  {
    text: "degree",
    value: "degree"
  }
];
const datums: DropdownItemProps[] = [
  {
    text: "Tokyo",
    value: "Tokyo"
  },
  {
    text: "WGS84",
    value: "WGS84"
  }
];

export function GeodeticInput(props: Props): React.ReactElement<Props> {
  return (
    <>
      <Dropdown
        fluid
        onChange={props.changeUnit}
        options={units}
        style={{ marginTop: "10px", marginBottom: "10px" }}
        text={props.unit}
        value={props.unit}
      />
      <Dropdown
        fluid
        onChange={props.onDatumChanged}
        options={datums}
        style={{ marginTop: "10px", marginBottom: "10px" }}
        text={props.datum}
        value={props.datum}
      />
    </>
  );
}
