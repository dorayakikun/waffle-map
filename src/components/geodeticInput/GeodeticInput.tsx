import { Select, Stack } from "@chakra-ui/react";
import * as React from "react";
export type Props = {
  unit: string;
  datum: string;
  onUnitChanged: (event: React.SyntheticEvent<HTMLElement>) => void;
  onDatumChanged: (event: React.SyntheticEvent<HTMLElement>) => void;
};

export function GeodeticInput(props: Props): React.ReactElement<Props> {
  return (
    <Stack spacing={3}>
      <Select onChange={props.onUnitChanged} size="md" value={props.unit}>
        <option value={"millisec"}>millisec</option>
        <option value={"degree"}>degree</option>
      </Select>
      <Select onChange={props.onDatumChanged} size="md" value={props.datum}>
        <option value={"Tokyo"}>Tokyo</option>
        <option value={"WGS84"}>WGS84</option>
      </Select>
    </Stack>
  );
}
