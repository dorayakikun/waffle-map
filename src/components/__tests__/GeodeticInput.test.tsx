import * as React from "react";
import * as render from "react-test-renderer";
import { GeodeticInput, Props } from "../GeodeticInput";

const props: Props = {
  changeUnit: () => {},
  datum: "WGS84",
  onDatumChanged: () => {},
  unit: "degree"
};

test("Should set props to GeodeticInput", () => {
  const tree = render.create(<GeodeticInput {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Should set props to GeodeticInput(datum is Tokyo)", () => {
  const datum = "Tokyo";
  const newProps = { ...props, datum };
  const tree = render.create(<GeodeticInput {...newProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Should set props to GeodeticInput(unit is millisec)", () => {
  const unit = "millisec";
  const newProps = { ...props, unit };
  const tree = render.create(<GeodeticInput {...newProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});
