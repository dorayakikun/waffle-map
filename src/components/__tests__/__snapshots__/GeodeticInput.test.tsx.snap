// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should set props to GeodeticInput 1`] = `
Array [
  <div
    aria-expanded={false}
    className="ui fluid dropdown"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    style={
      Object {
        "marginBottom": "10px",
        "marginTop": "10px",
      }
    }
    tabIndex={0}
  >
    <div
      aria-live="polite"
      className="text"
      role="alert"
    >
      degree
    </div>
    <i
      aria-hidden="true"
      className="dropdown icon"
      onClick={[Function]}
    />
    <div
      className="menu transition"
    >
      <div
        aria-checked={false}
        aria-selected={false}
        className="item"
        onClick={[Function]}
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          millisec
        </span>
      </div>
      <div
        aria-checked={true}
        aria-selected={true}
        className="active selected item"
        onClick={[Function]}
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          degree
        </span>
      </div>
    </div>
  </div>,
  <div
    aria-expanded={false}
    className="ui fluid dropdown"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    style={
      Object {
        "marginBottom": "10px",
        "marginTop": "10px",
      }
    }
    tabIndex={0}
  >
    <div
      aria-live="polite"
      className="text"
      role="alert"
    >
      WGS84
    </div>
    <i
      aria-hidden="true"
      className="dropdown icon"
      onClick={[Function]}
    />
    <div
      className="menu transition"
    >
      <div
        aria-checked={false}
        aria-selected={false}
        className="item"
        onClick={[Function]}
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          Tokyo
        </span>
      </div>
      <div
        aria-checked={true}
        aria-selected={true}
        className="active selected item"
        onClick={[Function]}
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          WGS84
        </span>
      </div>
    </div>
  </div>,
]
`;

exports[`Should set props to GeodeticInput(datum is Tokyo) 1`] = `
Array [
  <div
    aria-expanded={false}
    className="ui fluid dropdown"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    style={
      Object {
        "marginBottom": "10px",
        "marginTop": "10px",
      }
    }
    tabIndex={0}
  >
    <div
      aria-live="polite"
      className="text"
      role="alert"
    >
      degree
    </div>
    <i
      aria-hidden="true"
      className="dropdown icon"
      onClick={[Function]}
    />
    <div
      className="menu transition"
    >
      <div
        aria-checked={false}
        aria-selected={false}
        className="item"
        onClick={[Function]}
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          millisec
        </span>
      </div>
      <div
        aria-checked={true}
        aria-selected={true}
        className="active selected item"
        onClick={[Function]}
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          degree
        </span>
      </div>
    </div>
  </div>,
  <div
    aria-expanded={false}
    className="ui fluid dropdown"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    style={
      Object {
        "marginBottom": "10px",
        "marginTop": "10px",
      }
    }
    tabIndex={0}
  >
    <div
      aria-live="polite"
      className="text"
      role="alert"
    >
      Tokyo
    </div>
    <i
      aria-hidden="true"
      className="dropdown icon"
      onClick={[Function]}
    />
    <div
      className="menu transition"
    >
      <div
        aria-checked={true}
        aria-selected={true}
        className="active selected item"
        onClick={[Function]}
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          Tokyo
        </span>
      </div>
      <div
        aria-checked={false}
        aria-selected={false}
        className="item"
        onClick={[Function]}
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          WGS84
        </span>
      </div>
    </div>
  </div>,
]
`;

exports[`Should set props to GeodeticInput(unit is millisec) 1`] = `
Array [
  <div
    aria-expanded={false}
    className="ui fluid dropdown"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    style={
      Object {
        "marginBottom": "10px",
        "marginTop": "10px",
      }
    }
    tabIndex={0}
  >
    <div
      aria-live="polite"
      className="text"
      role="alert"
    >
      millisec
    </div>
    <i
      aria-hidden="true"
      className="dropdown icon"
      onClick={[Function]}
    />
    <div
      className="menu transition"
    >
      <div
        aria-checked={true}
        aria-selected={true}
        className="active selected item"
        onClick={[Function]}
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          millisec
        </span>
      </div>
      <div
        aria-checked={false}
        aria-selected={false}
        className="item"
        onClick={[Function]}
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          degree
        </span>
      </div>
    </div>
  </div>,
  <div
    aria-expanded={false}
    className="ui fluid dropdown"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    style={
      Object {
        "marginBottom": "10px",
        "marginTop": "10px",
      }
    }
    tabIndex={0}
  >
    <div
      aria-live="polite"
      className="text"
      role="alert"
    >
      WGS84
    </div>
    <i
      aria-hidden="true"
      className="dropdown icon"
      onClick={[Function]}
    />
    <div
      className="menu transition"
    >
      <div
        aria-checked={false}
        aria-selected={false}
        className="item"
        onClick={[Function]}
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          Tokyo
        </span>
      </div>
      <div
        aria-checked={true}
        aria-selected={true}
        className="active selected item"
        onClick={[Function]}
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          WGS84
        </span>
      </div>
    </div>
  </div>,
]
`;
